xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -2222,23 +2224,31 @@ bool CActiveAE::RunStages()
                 memcpy(viz->pkt->data[i], out->pkt->data[i], bytes);
               }
               viz->pkt->nb_samples = samples;
+              viz->pkt_start_offset = out->pkt_start_offset;
```

Comment:
> Think to found the main bug about.
wrong. this change doesn't have any impact on the current code because timestamps of audiostream are not used for viz. pkt_start_offset is set to 0 in ResampleBuffers

Reply:
The `timestamps` of given audio streams are always 0. 
Here it do mostly the same https://github.com/xbmc/xbmc/blob/master/xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp#L1960 to get a PTS about (but never called as 0 on `buf->timestamp`.

But it cannot understand why this is not needed, in the resample it should correspond to the data from the input where this `pkt_start_offset` is missing from the copy.

The error is then even without a resample, only this value is so small that it is not noticeable.

Reply:
Here about the current problem, with values prcessed inside:
```cpp
      int out_samples = m_resampler->Resample(m_planes,
                                              m_procSample->pkt->max_nb_samples - m_procSample->pkt->nb_samples,
                                              in ? in->pkt->data : NULL,
                                              in ? in->pkt->nb_samples : 0,
                                              m_resampleRatio);
      //----------> 1. out_samples = 2173
      //----------> 2. out_samples = 2205
      //----------> 3. out_samples = 2205
      // ... Stays the same
       
      // in case of error, trigger re-create of resampler
      if (out_samples < 0)
      {
        out_samples = 0;
        m_changeResampler = true;
      }
      
      m_procSample->pkt->nb_samples += out_samples;
      //----------> 1. m_procSample->pkt->nb_samples: 2173 = 0 + 2173
      //----------> 2. m_procSample->pkt->nb_samples: 2205 = 0 + 2205
      //----------> 3. m_procSample->pkt->nb_samples: 2205 = 0 + 2205
      // ... Stays the same

      busy = true;
      m_empty = (out_samples == 0);

      if (in)
      {
        if (!timestamp)
        {
          if (in->timestamp)
            m_lastSamplePts = in->timestamp;
          else
            in->pkt_start_offset = 0;
        }
        else
        {
          m_lastSamplePts = timestamp;
          // Here how CPU clock brings wrong values about "timestamp" and destroy use
          //----------> 1. 16610 (correct 50 jump)
          //----------> 2. 16710 (goes 100 one step time overjumped, comes much times)
          //----------> 3. 16760 (correct 50 jump)
          //----------> ... ~50 on step (correct)
          //----------> 7. 16961 (correct 50 jump)
          //----------> 8. 17412 (goes 451 forward, also much times)
          //----------> 9. 17512 (goes 100 one step time overjumped, comes much times)
          //----------> 10. 17561 (correct 50 jump)
          // This time jumps always continued inside.
          
          in->pkt_start_offset = 0;
        }

        // pts of last sample we added to the buffer
        m_lastSamplePts += (in->pkt->nb_samples-in->pkt_start_offset) * 1000 / m_format.m_sampleRate;
        // Here where "pkt_start_offset" is always "0" (set short before) and the timestamp from input becomes always 400 added (where destroy the show time)
        // as a Paket have 50 in his size the audio engine must process additional 8 until it reachs the first for give to addon
        //----------> 1.  m_lastSamplePts: 17010 = 16610 + (17640 - 0) * 1000 / 44100
        //----------> 2.  m_lastSamplePts: 17110 = 16710 + (17640 - 0) * 1000 / 44100 <!-----
        //----------> 3.  m_lastSamplePts: 17160 = 16760 + (17640 - 0) * 1000 / 44100
        //----------> 4.  m_lastSamplePts: 17210 = 16810 + (17640 - 0) * 1000 / 44100
        //----------> 5.  m_lastSamplePts: 17260 = 16860 + (17640 - 0) * 1000 / 44100
        //----------> 6.  m_lastSamplePts: 17310 = 16910 + (17640 - 0) * 1000 / 44100
        //----------> 7.  m_lastSamplePts: 17361 = 16961 + (17640 - 0) * 1000 / 44100
        //----------> 8.  m_lastSamplePts: 17812 = 17412 + (17640 - 0) * 1000 / 44100 <!-----
        //----------> 9.  m_lastSamplePts: 17912 = 17512 + (17640 - 0) * 1000 / 44100 <!-----
        //----------> 10. m_lastSamplePts: 17961 = 17561 + (17640 - 0) * 1000 / 44100
      }

      // calculate pts for last sample in m_procSample
      int bufferedSamples = m_resampler->GetBufferedSamples();
      //----------> 1. bufferedSamples = 32
      //----------> 2. bufferedSamples = 32
      //----------> 3. bufferedSamples = 32
      // ... Stays the same
      
      m_procSample->pkt_start_offset = m_procSample->pkt->nb_samples;
      //----------> 1. m_procSample->pkt_start_offset = 2173 (m_procSample->pkt->nb_samples)
      //----------> 2. m_procSample->pkt_start_offset = 2205 (m_procSample->pkt->nb_samples)
      //----------> 3. m_procSample->pkt_start_offset = 2205 (m_procSample->pkt->nb_samples)
      // ... Stays the same
      
      m_procSample->timestamp = m_lastSamplePts - bufferedSamples * 1000 / m_format.m_sampleRate;
      //----------> 1. m_procSample->timestamp: 17010 = 17010 - 32 * 1000 / 44100
      //----------> 2. m_procSample->timestamp: 17110 = 17110 - 32 * 1000 / 44100 <!-----
      //----------> 3. m_procSample->timestamp: 17160 = 17160 - 32 * 1000 / 44100
      //----------> 4. m_procSample->timestamp: 17210 = 17210 - 32 * 1000 / 44100
      //----------> 5. m_procSample->timestamp: 17260 = 17260 - 32 * 1000 / 44100
      //----------> 6. m_procSample->timestamp: 17310 = 17310 - 32 * 1000 / 44100
      //----------> 7. m_procSample->timestamp: 17361 = 17361 - 32 * 1000 / 44100
      //----------> 8. m_procSample->timestamp: 17812 = 17812 - 32 * 1000 / 44100 <!-----
      //----------> 9. m_procSample->timestamp: 17912 = 17912 - 32 * 1000 / 44100 <!-----
      //----------> 10. m_procSample->timestamp: 17961 = 17961 - 32 * 1000 / 44100
      
```
Is from "CActiveAEBufferPoolResample::ResampleBuffers"

Where should be the place to correct the the wrongly in future shown timestamp without use of `pkt_start_offset`?

Reply:
What also works without set of `pkt_start_offset`in AudioEngine, to use for `m_lastSamplePts` the `m_procSample->pkt` instead of `in->pkt`
```cpp
        // pts of last sample we added to the buffer
        int64_t samples = m_procSample->pkt->nb_samples-m_procSample->pkt_start_offset;
        if (samples > 0)
          m_lastSamplePts += samples * 1000 / m_format.m_sampleRate;
```

Is that more correct as the note `// pts of last sample we added to the buffer` also rever to avbout add to buffer where is `m_procSample`?

Reply:
Before blaming code you obviously do not fully understand and trying to change things in a try and error fashion, you should really try to answer this questions: Why does it work with normal sample rates and why do you observe issues with high sample rates. 

```
         m_lastSamplePts = timestamp;
          // Here how CPU clock brings wrong values about "timestamp" and destroy use
          //----------> 1. 16610 (correct 50 jump)
          //----------> 2. 16710 (goes 100 and skipped a step, comes often)
```

Jumps in timestamp here are either a result of GetDelay of audio sink or AEDelayStatus::maxcorrection prevents correction of delay with current time (maxcorrection might be zero because it is not set)